### PR TITLE
Implement character customisation closes #46

### LINKED ISSUE #46: Costumes
**User Story**
As a gamer, I'd like to be able to change my appearance, so that my character is more personalised

**Acceptance Criteria**

- Have different costumes for the main character
- Want to be able to select these at the begining or throughout the game

**Notes**

Costumes could just be colour changes from the orginal sprite

---

**Why is this feature needed? Please describe the problem your requested feature wants to solve**
Character always looks the same

**Describe the solution you'd like**
User has a choice of character looks

### PR DESCRIPTION
**Pull request by Matteas-Eden**
GitHub Issue (If applicable): #46 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

There is no functionality for customising the colour of the character sprite.

## What is the new behavior?

A menu has been added (after character creation, before ability scores) that allows a player to customise the colours of:
* Hair
* Skin
* Armour
* Clothes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [ ] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information

The current implementation simply uses hue rotation to achieve the customisation, and is therefore limited slightly in the range of colours that can be displayed.

Internal Issue (If applicable): Closes #46

### CODE REVIEW

**Review by Pyxxil — CHANGES_REQUESTED** [review]
Looks mostly good from looking at the code (I'll test it properly later), this is just a small implementation detail.

File: src/features/dialog-manager/dialogs/character-customisation/index.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                                <ColourPicker onChange={setArmourColour} />
+                            </div>
+                        </div>
+                        <div className="flex-row character-customisation__option">
+                            <p className="flex-column character-customisation__label">
+                                Clothes
+                            </p>
+                            <div className="character-customisation__picker">
+                                <ColourPicker onChange={setClothesColour} />
+                            </div>
```
Comment (Pyxxil) [inline src/features/dialog-manager/dialogs/character-customisation/index.js]: Could we modify this similar to how we did the Select Buttons and Ability Scores?

**Review by Pyxxil — CHANGES_REQUESTED** [review]
Sorry for the double up my internet is trash and this didn't send, this refers to the previous change request

File: src/features/player/index.js [hunk 2]
```diff
@@ -10,8 +10,16 @@ import PlayerDeath from './assets/player-death.mp3';
...
+
+import PlayerHair from './assets/player-hair.png';
+import PlayerEyes from './assets/player-eyes.png';
+import PlayerSkin from './assets/player-skin.png';
+import PlayerArmour from './assets/player-armour.png';
+import PlayerClothes from './assets/player-clothes.png';
+import PlayerOutline from './assets/player-outline.png';
+
 import { ANIMATION_SPEED, SPRITE_SIZE } from '../../config/constants';
+import createFilterFromHexCode from '../../utils/create-filter-from-hex-code';
```
Comment (Pyxxil) [inline src/features/player/index.js]: This is imported, but not used.

File: src/utils/create-filter-from-hex-code.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    let b = parseInt(result[3], 16);
+
+    let hsv = rgbToHsv(r, g, b);
+
+    hsv[0] *= 360;
+
+    let filter = `hue-rotate(${hsv[0]}deg) saturate(${hsv[1]}) brightness(${hsv[2]})`;
+
+    return filter;
+}
```
Comment (Pyxxil) [inline src/utils/create-filter-from-hex-code.js]: This entire function isn't used (or, at least, you commented out its use). So do we need it?

File: src/features/player/index.js [hunk 4]
```diff
@@ -50,13 +58,87 @@ class Player extends Component {
...
+                    SPRITE_SIZE
+                );
+                // Draw player outline
+                ctx.filter = 'none';
                 ctx.drawImage(
-                    this.sprite,
+                    this.sprite.outline,
                     frame * SPRITE_SIZE,
                     spriteLine,
                     SPRITE_SIZE,
```
Comment (Pyxxil) [inline src/features/player/index.js:144]: Would it be possible to make this into a loop?

File: src/features/player/index.js [hunk 5]
```diff
@@ -50,13 +58,87 @@ class Player extends Component {
...
+                    spriteLine,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE,
+                    0,
+                    0,
+                    SPRITE_SIZE,
+                    SPRITE_SIZE
+                );
+                // Draw player outline
+                ctx.filter = 'none';
```
Comment (Pyxxil) [inline src/features/player/index.js]: Would it be possible to turn all of these (and possibly the one below them) into a loop?

Comment (Matteas-Eden) [reply src/features/player/index.js]: Previous attempts of mine to iterate over `this.sprite` weren't successful. Do you mind suggesting a way to loop over it?

Comment (Pyxxil) [reply src/features/player/index.js]: Does `for (var key in this.sprite)` not work? (You'd have to special case the 'onload' key). But if it doesn't work easily, then it's not a big deal.

File: src/components/colour-picker/index.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        this.setState({ displayColorPicker: !this.state.displayColorPicker });
+    };
+
+    handleClose = () => {
+        this.setState({ displayColorPicker: false });
+    };
+
+    handleChange = color => {
+        this.setState({ color: color.hsl });
+        console.log(color.hsl.h);
```
Comment (Pyxxil) [inline src/components/colour-picker/index.js]: Probably don't need this line

**Review by Pyxxil — DISMISSED** [review]
Looks good. Well done on finally getting it done.

**Review by Pyxxil — APPROVED** [review]
LGTM

**Pull request merged**